src/x86_64/unwind_rule.rs

```diff
@@ -19,6 +20,120 @@ pub enum UnwindRuleX86_64 {
     },
     /// (sp, bp) = (bp + 16, *bp)
     UseFramePointer,
+    /// sp = sp + 8 * (offset + register count)
+    /// This supports the common case of pushed callee-saved registers followed by a stack
+    /// allocation. Up to 8 registers can be stored, which covers all callee-saved registers (aside
+    /// from RSP which is implicit).
+    ///
+    /// The registers are stored in a separate compressed ordering to facilitate restoring register
+    /// values if desired. If not for this we could simply store the total offset.
+    OffsetSpAndPopRegisters(UnwindRuleOffsetSpAndPopRegisters),
+}
+
+#[derive(Default, Clone, Copy, Debug, PartialEq, Eq)]
+pub struct UnwindRuleOffsetSpAndPopRegisters {
+    pub sp_offset_by_8: u16,
+    registers: u16,
+}
+
+pub enum OffsetOrPop {
+    None,
+    OffsetBy8(u16),
+    Pop(Reg),
+}
```

Comment:
Aside from a few things desired to be added, including frame / base pointer support, there should be a way to express Noop in case of "save register" codes (these do not alter stack or frame pointer, except when restoring frame pointer). Right now any frames that have a "save" code is disqualified from the fast path.